grove/connectors/fleetdm/vulnerability_logs.py

```diff
@@ -0,0 +1,124 @@
+from grove.exceptions import NotFoundException
+
+
+class Connector(BaseConnector):
+    NAME = "fleetdm_vulnerability_logs"
+    POINTER_PATH = "updated_at"
+
+
+    @property
+    def jmespath_queries(self):
+        """Fetches the parameters for the jmespath filters of the data response.
+
+        Jmespath query language is defined and can be tested at https://jmespath.org/
+        This allows you to configure what data to include or filter from the FleetDM response.
+
+        An example is:
+        "jmespath_queries": "{hostname:hostname,updated_at:updated_at,software_updated_at:software_updated_at,
+        uuid:uuid,hardware_serial:hardware_serial,computer_name:computer_name,
+        os_version:osversion,software:software[?vulnerabilities].{name:name,
+        version:version,source:source,vulnerabilities:vulnerabilities[?cvss_score].{cve:cve,cvss_score:cvss_score,
+        epss_probability:epss_probability,cisa_known_exploit:cisa_known_exploit,resolved_in_version:resolved_in_version}}}"
+
+        This returns the following structure of data about each host:
+        {
+        updated_at:
+        software_updated_at:
+        hostname:
+        uuid:
+        hardware_serial:
+        computer_name:
+        os_version:
+        software:{ (filtered to only software with vulnerabilities which have a CVSS score)
+            name:
+            version:
+            source:
+            vulnerabilities:{
+                cve:
+                cvss_score:
+                epss_probability:
+                cisa_known_exploit:
+                resolved_in_version:
+            }
+        }
+
+        :return: A string of the Jmespath response that should define the JSON object to return.
+        Default is *, the full set of JSON response
+        """
+        try:
+            p = self.configuration.jmespath_queries
+        except AttributeError:
+            return "*"
+
+        return p
+
+    @property
+    def params(self):
+        """Fetches the parameters for the API call.
+
+        This is used to set what parameters to use in the API call.
+
+        :return: The dict of params defined in the connector configuration.
+        """
+        try:
+            p = self.configuration.params
+        except AttributeError:
+            return None
+
+        return p
+    
+    @property
+    def api_uri(self):
+        """The URI for the API call
+        For example: https://panel.fleetdm.example.com
+        """
+        try:
+            p = self.configuration.api_uri
+        except AttributeError:
+            return None
+
+        return p
+
+    def collect(self):
+        """Collects all hosts from the FleetDM API.
+        """
+        client = Client(
+            token=self.key,
+            params=self.params,
+            jmespath_queries=self.jmespath_queries,
+            api_uri=self.api_uri
+            )
+        
+
+        # We do a full load of hosts on each run as there's no good way to determine if
```

Comment:
With the move to using `operation` will this change and allow for use of the `pointer` to ensure this is only new data per `operation` type?

I ask, as the default execution time of Grove is quite aggressive, so loading this data every 10-minutes or so may be quite a bit of data depending on the size of the fleet :)